ably/realtime_channel.go

```diff
@@ -451,29 +320,27 @@ func (c *RealtimeChannel) notify(msg *proto.ProtocolMessage) {
 	case proto.ActionAttached:
 		c.state.syncSet(StateChanAttached, nil)
 		c.queue.Flush()
+		c.Presence.mu.Lock()
+		c.Presence.syncStart()
+		if msg.Flags&1 == 0 {
+			c.Presence.syncEnd()
+		}
+		c.Presence.mu.Unlock()
 	case proto.ActionDetached:
 		c.state.syncSet(StateChanDetached, nil)
+	case proto.ActionSync:
+		syncSerial := ""
+		if i := strings.IndexRune(msg.ChannelSerial, ':'); i != -1 {
+			syncSerial = msg.ChannelSerial[i+1:]
+		}
+		c.Presence.handle(msg, syncSerial)
 	case proto.ActionPresence:
-		// TODO realtime presence
+		c.Presence.handle(msg, "")
```

Comment:
not sure I like the name handle, what exactly is that method doing?  Could you name it more descriptively please?